packages/editor/src/components/block-list/block.js

```diff
@@ -672,14 +675,19 @@ const applyWithSelect = withSelect( ( select, { clientId, rootClientId, isLargeV
 		mode: getBlockMode( clientId ),
 		isSelectionEnabled: isSelectionEnabled(),
 		initialPosition: getSelectedBlocksInitialCaretPosition(),
-		isEmptyDefaultBlock: block && isUnmodifiedDefaultBlock( block ),
-		isPreviousBlockADefaultEmptyBlock: previousBlock && isUnmodifiedDefaultBlock( previousBlock ),
+		isEmptyDefaultBlock: blockName && isUnmodifiedDefaultBlock( { name: blockName, attributes: blockAttributes } ),
+		isPreviousBlockADefaultEmptyBlock: previousBlockClientId && isUnmodifiedDefaultBlock( {
```

Comment:
My concern from https://github.com/WordPress/gutenberg/pull/11811#discussion_r233105376 still holds true here.

Reply:
Not strictly related to the pull request, but: Why do we care about not showing a sibling inserter if the previous block is an empty default block? I might do some separate exploring 😄

Reply:
This was @jasmussen's proposal I think. I also would prefer removing it if it doesn't harm UX

Reply:
Just to be sure I understand this correctly: this rule hides the sibling inserter between an empty paragraph block and another block?

If correct, the reasoning for removing that likely stems back to when the sibling inserter click area spanned the full width and it was way too easy to invoke it. Given recent changes to this hit area and the sibling inserter behavior itself, totally okay to revisit and see if it's a good experience.